rest/data/project_settings.go

```diff
@@ -275,7 +275,7 @@ func SaveProjectSettingsForSection(ctx context.Context, projectId string, change
 		}
 
 		if mergedSection.Enabled {
-			if mergedSection.Branch == "" {
+			if mergedSection.Branch == "" && !isRepo {
 				return nil, errors.New("branch not set on enabled repo")
```

Comment:
maybe we should swap this to "enabled project" so this doesn't seem like an error. Can we also put a comment here to say branch isn't defined for repos so we remember why we special-cased this?